x-pack/auditbeat/processors/sessionmd/timeutils/time.go

```diff
@@ -0,0 +1,93 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+//go:build linux
+
+package timeutils
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/prometheus/procfs"
+	"github.com/tklauser/go-sysconf"
+)
+
+var (
+	bootTime       time.Time
+	ticksPerSecond uint64
+	initError      error
+	once           sync.Once
+)
+
+func initialize() {
+	var err error
+	bootTime, err = getBootTime()
+	if err != nil {
+		initError = err
+		return
+	}
+
+	ticksPerSecond, err = getTicksPerSecond()
+	if err != nil {
+		initError = err
+	}
+}
```

Comment:
https://pkg.go.dev/sync#OnceValues would be perfect for these two one-time functions.

```suggestion
var (
	getBootTimeOnce       = sync.OnceValues(getBootTime)
	getTicksPerSecondOnce = sync.OnceValues(getTicksPerSecond)
)
```

```go
	// Example Usage
	ticksPerSecond, err := getTicksPerSecondOnce()
	if err != nil {
		return 0
	}
```

Reply:
Thanks, I didn't know about that, changed